### ISSUE TITLE
Severe performance degradation

### ISSUE THREAD

**Issue by vmarkovtsev** [src-d/enry#65]
Something happened and now enry works even slower than linguist. Seriously, testing on src-d/ast2vec:

```
time ./enry .
{
  "Python": [
    "ast2vec/__init__.py",
    "ast2vec/__main__.py",
    "ast2vec/df.py",
    "ast2vec/dump.py",
    "ast2vec/enry.py",
    "ast2vec/id2vec.py",
    "ast2vec/id_embedding.py",
    "ast2vec/meta.py",
    "ast2vec/model.py",
    "ast2vec/nbow.py",
    "ast2vec/publish.py",
    "ast2vec/repo2base.py",
    "ast2vec/repo2coocc.py",
    "ast2vec/repo2nbow.py",
    "ast2vec/swivel.py",
    "ast2vec/tests/__init__.py",
    "ast2vec/tests/models.py",
    "ast2vec/tests/test_dump.py",
    "ast2vec/tests/test_enry.py"
  ],
  "Text": [
    "requirements.txt"
  ]
}

real	0m14.339s
user	0m10.740s
sys	0m1.972s
```

```
time linguist
100.00% Python

real	0m2.192s
user	0m2.016s
sys	0m0.104s
```

**Comment by vmarkovtsev** [comment]
Recent HEAD

```
time ./enry .
95.00%	Python
5.00%	Text

real	0m4.098s
user	0m1.064s
sys	0m1.984s
```

**Comment by smola** [comment]
It seems you have huge files in that directory.
In simple-linguist v1.0.0, the CLI did bypass loading the file if it could be classified just with filename and extension: https://github.com/src-d/enry/blob/ec9c23e4118b2ec834c41592c6b98e2099c54426/cmd/slinguist/main.go#L41
This was removed in later version (including enry). But we should bring this optimization back.

**Comment by vmarkovtsev** [comment]
Yep, I have 1GB file named "id2vec_1MM_serial.asdf", which starts as a regular YAML but has a binary ending.

**Comment by abeaumont** [comment]
This should be solved by #67
